packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -182,11 +195,33 @@ export default class Sayt extends LitElement {
       this.hideSayt();
       return;
     }
-
     this.requestSaytAutocompleteTerms(query, searchbox);
     this.requestSaytProducts(query, searchbox);
   }
 
+  /**
+   * Debounces the function that is passed in the callback paramater.
+   * Customizable delay based on debounceTime attribute on Sayt element.
+   *
+   * @param callback The function to debounce.
+   */
+  getDebounce(callback: any) {
+    return debounce(
+      callback,
+      this.debounce,
+      false
+    );
+  }
+
+  /**
+   * Dispatches getDebounce for each of the methods to be debounced.
+   * Is triggered once in constructor and when the debounce attribute is changed.
+   */
+  setDebounce() {
+    this.requestSaytAutocompleteTerms  = this.getDebounce(this.requestSaytAutocompleteTerms);
```

Comment:
What happens to the debounce when changing the wait time after it has already been set? The assumption is that `this.requestSaytAutocompleteTerms` will hold reference to a debounced method when being passed to `getDebounce()` a second time. 

Ex. First debounce being 300ms, second debounce setting is 400ms, resulting in a 700ms debounce. 

If this acts like so, a possible solution would be to name the original `requestSaytAutocompleteTerms()` and `requestSaytProducts` in a pseudo private method (ie. `_requestSaytAutocompleteTerms()`) and then continuing to assign the debounced method to the same property.